### PR TITLE
Updated ExpandTensorShapes to work on SCF operations

### PR DESCRIPTION
**Pull request by rsuderman**
Part of making Flow transforms support SCF requires that the expand tensor shapes pass can recurse in / expand the shape operations.

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp [hunk 1]
```diff
@@ -364,28 +392,7 @@ static void expandCallOp(mlir::func::CallOp op, IndexSet &indexSet,
   // Insert shape ties on results that we are sinking across the call edge.
   // The hope is that by moving the ties here we can fold with queries inside of
   // this function.
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp]: This should be moved to function doc?

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp [hunk 2]
```diff
@@ -484,6 +491,67 @@ static void expandSelectOp(mlir::arith::SelectOp op, IndexSet &indexSet,
...
+  OpBuilder builder(op);
+  auto operands = expandOperands(op.getLoc(), op.getOperands(), tensorDimMap,
+                                 indexSet, builder);
+  auto resultTypes = expandTypes(op.getResultTypes());
+
+  auto newOp = builder.create<scf::WhileOp>(op.getLoc(), resultTypes, operands,
+                                            /*beforeBody*/ nullptr,
+                                            /*afterBody*/ nullptr);
+
+  IRMapping mapping;
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp]: unused

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp [hunk 3]
```diff
@@ -484,6 +491,67 @@ static void expandSelectOp(mlir::arith::SelectOp op, IndexSet &indexSet,
...
+
+static void expandIfOp(mlir::scf::IfOp op, ExpandedGlobalMap &globalMap,
+                       IndexSet &indexSet, TensorDimMap &tensorDimMap) {
+  OpBuilder builder(op);
+  auto resultTypes = expandTypes(op.getResultTypes());
+
+  auto newOp = builder.create<scf::IfOp>(
+      op.getLoc(), resultTypes, op.getOperand(), op.elseBlock() != nullptr);
+
+  IRMapping mapping;
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp]: unused

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.cpp [hunk 4]
```diff
@@ -224,6 +224,7 @@ void buildFlowTransformPassPipeline(OpPassManager &passManager,
       // compute op into the dispatch region, so that we can run additional
       // transformations afterwards with a simple region and without bothering
       // producers.
+      .addPass(IREE::Flow::createTopLevelSCFToCFGPass)
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.cpp:173]: Why do we move it between `FormScalarDispatchPass` and `FormDispatchRegionsPass`? I thought that it should happen maybe right before `createFusionOfTensorOpsPass`?

Comment (rsuderman) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.cpp:173]: `FormDispatchRegionsPass` does not support `scf` operations yet. I was going to continue pushing the `scf` support forward pass-by-pass.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp (hunk 1 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp]: Done.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp (hunk 3 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp]: Done.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp (hunk 2 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/ExpandTensorShapes.cpp]: Done.

**Review by hanhanW — APPROVED** [review]

**Pull request merged**